wait/host_port_test.go

```diff
@@ -1,18 +1,25 @@
-package wait
+package wait_test
```

Comment:
suggestion: no need to make this change. We will do that separately as it’s just adding unnecessary changes

Reply:
If we don't do this, then we'll have to export `mockStrategyTarget` as it is `wait_test` while our tests were originally in `wait`. So, I moved `host_port_test.go` to `wait_test` and exported the necessary variables from `host_port.go`.

Reply:
You never want to export functions to test like this. There are two options, do a end to end which exercises these cases or a seperate file which which is in the `wait` packages.

Reply:
As it uses the logger you can capture the output in the test and assert the content. e.g.
```go
func MyTest(t *testing.T) {
	oldWriter := log.Writer()
	var buf bytes.Buffer
	log.SetWriter(&buf)
	t.Cleanup(func() {
		log.SetWriter(oldWriter)
	})

	// Run the test
	require.Contains(t, buf.String(), "Shell not found in container")
}
```

Reply:
@stevenh made the changes. PTAL. I've merged internal check tests with the existing tests.
Also, I've reverted back to `MockStrategyTarget` as it will lead to unnecessary code changes for importing certain variables and methods. Please handle `MockStrategyTarget` -> `mockStrategyTarget` for this test in your draft PR.